crates/trigger-http/src/server.rs

```diff
@@ -138,12 +148,46 @@ impl<F: RuntimeFactors> HttpServer<F> {
 
     /// Serve incoming requests over the provided [`TcpListener`].
     pub async fn serve(self: Arc<Self>) -> anyhow::Result<()> {
-        let listener = TcpListener::bind(self.listen_addr).await.with_context(|| {
-            format!(
-                "Unable to listen on {listen_addr}",
-                listen_addr = self.listen_addr
-            )
-        })?;
+        let listener: TcpListener = match TcpListener::bind(self.listen_addr).await {
+            Ok(listener) => listener,
+            Err(err) => {
+                if self.find_free_port && err.kind() == ErrorKind::AddrInUse {
+                    let mut found_listener = None;
+                    let mut addr = self.listen_addr;
+                    for _ in 1..=9 {
+                        addr.set_port(addr.port() + 1);
+
+                        if !addr.port() == u16::MAX {
```

Comment:
The `!` operator here is bitwise complement.  Is that what you intended?

Reply:
I missed that, was initially trying a `while` loop. Would fix and push a fix now